include/deal.II/base/aligned_vector.h

```diff
@@ -1553,6 +1553,10 @@ AlignedVector<T>::replicate_across_communicator(const MPI_Comm &   communicator,
   used_elements_end      = elements.get() + array_size;
   allocated_elements_end = used_elements_end;
 
+  // Make sure the pointers are correctly set for all processes before anyone
+  // accesses the shared data
+  MPI_Barrier(communicator);
```

Comment:
I don't think the explanation here is correct. I would say that the barrier needs to be semantically after the `Utilities::MPI::broadcast(shmem_group_communicator, ...)`, since I think the data might not be ready yet and not the pointers.

Reply:
I tried all places between steps now and it really is between step-6 and step-7. I am pretty sure that the root process might not yet have copied the data before other processes try to access and modify it.